### ISSUE TITLE
feat: vision model embedding extraction for time series images

### ISSUE THREAD

**Issue by thomaschi78** [drumtorben/polars-ts#103]
## Context

Once time series are converted to images (recurrence plots, GAF, MTF, spectrograms), **pretrained vision models** can extract rich feature embeddings without any training. Recent work shows that ViT/CLIP/ResNet features on time series images achieve competitive clustering results, effectively leveraging the massive pretraining of vision foundation models.

**Tier: 2-3 (moderate effort)** — adapter that takes polars-ts image outputs and extracts embeddings via torchvision or CLIP.

**References:**
- Dosovitskiy, A. et al. (2021). *An Image is Worth 16x16 Words: ViTs*. ICLR.
- Radford, A. et al. (2021). *Learning Transferable Visual Models From Natural Language Supervision* (CLIP). ICML.
- Li, Z. et al. (2023). *Time Series as Images: ViT for Irregularly Sampled Time Series*. NeurIPS.

## Proposed API

```python
from polars_ts import to_gasf, extract_vision_embeddings

# Step 1: Convert series to images
images = to_gasf(df)

# Step 2: Extract embeddings from a pretrained vision model
embeddings = extract_vision_embeddings(
    images,
    model="resnet18",  # or "vit_b_16", "clip"
    layer="avgpool",   # which layer to extract from
)
# Returns: pl.DataFrame with [unique_id, emb_0, emb_1, ..., emb_d]

# Step 3: Cluster in embedding space
from polars_ts import kmedoids
labels = kmedoids(embeddings, k=3, method="euclidean")
```

### Task 1: Implement `extract_vision_embeddings`
- **File:** `polars_ts/imaging/embeddings.py` (new)
- Load pretrained model via `torchvision.models` or `transformers`
- Convert grayscale TS images to 3-channel (duplicate or colormap)
- Resize to model input size (224×224)
- Forward pass → extract intermediate layer activations
- Return polars DataFrame with one row per series
- **Acceptance criteria:**
  - [ ] Supports ResNet18/50, ViT-B/16
  - [ ] Returns DataFrame with `[id_col, emb_0, ..., emb_d]`
  - [ ] Graceful ImportError when torch not installed
  - [ ] Deterministic (eval mode, no dropout)
- **Scope:** Medium

### Task 2: Optional CLIP support
- Use `transformers` CLIPModel for joint vision-language embeddings
- **Scope:** Small (add to existing file)

### Task 3: Register and test (with importorskip)
- **Scope:** Small

## Dependencies
- `torch` and `torchvision` (optional)
- `transformers` (optional, for CLIP)
- Depends on #101 (recurrence plots) or #102 (GAF/MTF) for image generation

## Verification
```bash
uv run pytest tests/imaging/test_embeddings.py -v
```
